File: ch06/CS/Data/Data/AddProductForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Data
{
    public partial class AddProductForm : Form
    {
        private Category category;

        public AddProductForm(Category category)
        {
            InitializeComponent();

            this.category = category;
            categoryTextBox.Text = category.CategoryName;
        }

        private Product product;

        public Product Product
        {
            get
            {
                var result = new Product(0,
                    nameTextBox.Text,
                    category.CategoryID,
                    quantityPerUnitTextBox.Text,
                    Convert.ToDecimal(priceTextBox.Text),
                    Convert.ToInt32(stockTextBox.Text),
                     Convert.ToInt32(unitsOnOrderTextBox.Text),
                    discontinuedCheckBox.Checked);
                return result;
            }
        }

    }
}

File: ch07/VB/Data/Data/AddProductForm.vb
Public Class AddProductForm
    Private _category As Category

    Public Sub New(category As Category)

        ' This call is required by the designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.

        _category = category
        CategoryTextBox.Text = _category.CategoryName
    End Sub

    Public ReadOnly Property Product As Product
        Get
            Dim result As New Product(0, NameTextBox.Text, _category.CategoryID, QuantityPerUnitTextBox.Text, PriceTextBox.Text, StockTextBox.Text, UnitsOnOrderTextBox.Text, DiscontinuedCheckBox.Checked)
            Return result
        End Get
    End Property

End Class